// prettier-plugin-django-template/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.options = exports.printers = exports.parsers = exports.languages = void 0;
const parser_1 = require("./parser");
const printer_1 = require("./printer");
const PLUGIN_KEY = "django-template";
exports.languages = [
    {
        name: "DjangoTemplate",
        parsers: [PLUGIN_KEY],
        extensions: [".django.html", ".djhtml", ".html"],
        vscodeLanguageIds: ["django-html"],
    },
];
exports.parsers = {
    [PLUGIN_KEY]: {
        astFormat: PLUGIN_KEY,
        parse: parser_1.parse,
        locStart: (node) => node.index,
        locEnd: (node) => node.index + node.length,
    },
};
exports.printers = {
    [PLUGIN_KEY]: {
        print: printer_1.print,
        embed: printer_1.embed,
        getVisitorKeys: printer_1.getVisitorKeys,
    },
};
exports.options = {
    quoteAttributes: {
        type: "boolean",
        category: PLUGIN_KEY,
        default: true,
        description: "Surrounds the value of html attributes with quotes.",
    },
};
